src/constraints/StringConstraints.ts

```diff
@@ -2,8 +2,22 @@ import { ConstraintError } from '../lib/errors/ConstraintError';
 import { Result } from '../lib/Result';
 import type { IConstraint } from './base/IConstraint';
 import { Comparator, eq, ge, gt, le, lt, ne } from './util/operators';
+import net from 'node:net';
 
-export type StringConstraintName = `s.string.length${'Lt' | 'Le' | 'Gt' | 'Ge' | 'Eq' | 'Ne'}`;
+export type StringConstraintName =
+	| `s.string.${`length${'Lt' | 'Le' | 'Gt' | 'Ge' | 'Eq' | 'Ne'}` | 'regex' | 'url' | 'uuid' | 'email'}`
+	| StringIpName;
+
+export type StringIpName = `s.string.ip${'v4' | 'v6' | ''}`;
```

Comment:
Shouldn't exist. Anything using this should instead cast to `StringConstraintName`.